pkg/ops/miningManager.go

```diff
@@ -433,3 +335,22 @@ func (mgr *MiningMgr) profit() (int64, error) {
 
 	return profitPercent, nil
 }
+
+// Stop will take care of stopping the dataserver component.
+func (mgr *MiningMgr) Stop() {
+	level.Info(mgr.logger).Log("msg", "shutting down data server...")
+	mgr.close()
+	cnt := 0
+	for {
+		time.Sleep(500 * time.Millisecond)
+		cnt++
+		if !mgr.Running {
+			break
+		}
+		if cnt > 60 {
+			level.Warn(mgr.logger).Log("msg", "expected miner to stop by now, Giving up...")
+			return
+		}
+	}
```

Comment:
If we are using context properly this shouldn't be needed. I know that this is the same in few other places, but lets at least not add new ones.

Again if we use context properly we shouldn't not need the `mgr.Running` variable at all.

Reply:
I felt the same, but I thought maybe this needed to inform the user. so I remove them here and in other places.